packages/data/src/world/mm/stone_tower_temple.yml

```diff
@@ -83,7 +83,7 @@
 "Stone Tower Temple Water Room":
   dungeon: ST
   exits:
-    "Stone Tower Temple Center": "has_mask_zora || (soul_enemy(SOUL_ENEMY_DEXIHAND) && (short_hook_anywhere || (has_tunic_zora && has_iron_boots))) || (has_tunic_zora && has_iron_boots && (can_hookshot || short_hook_anywhere))"
+    "Stone Tower Temple Center": "has_mask_zora || (soul_enemy(SOUL_ENEMY_DEXIHAND) && (short_hook_anywhere || can_dive_small || (has_tunic_zora && has_iron_boots))) || (has_tunic_zora && has_iron_boots && (can_hookshot || short_hook_anywhere))"
```

Comment:
gonna need to know how much timer expires when doing this and the above one. If it's more than 34s with the 80 frames, it's gonna need to hard require tunic, otherwise should probably include tunic if it's too close to 34s like with GBT's Baba room.

Reply:
It takes less than 10 seconds from when the timer begins to get through the tunnel.